DAC (Team. Korea)

junbeomic opened the pull request:
request for DAC
it includes schematics of blocks used in DAC and basic things of layout.

bmurmann reviewed (changes requested):
Some of the symbol paths seem to contain an absolute component, which makes the schematics non-portable. For example: designs/digital_cell/local_decoder.sym -> "designs" does not exist in the repo tree and hence will only resolve on your personal system.

junbeomic commented:
I fixed symbol paths and added a new folder including standard cell symbols.
I will upload overall schematics as soon as possible and layouts.

bmurmann commented on DAC/xschem/analog_cell/dac_current_unit.sch:
Why are some VSS nets connected to "0" instead of VSS?
The transistor sizes are very small. What are the nominal currents? These should be annotated in the schematic as text.
I would use minimum L for XM3 and XM5 and perhaps a bit longer L for XM4 and XM6 (these are the matching sensitive devices in the cell).
The cascode biasing approach using a stack of two diodes uses up a lot of headroom. While it may not be absolutely necessary, a high-swing biases approach would be preferred.
The sheet is very large, but the circuit is small. Make it easier to see the circuit for anyone opening this sch.

bmurmann commented on DAC/xschem/analog_cell/test_dac_lsb_matrix.sch:
180MCU_REF_7T5V0/spice/gf180mcu_fd_sc_mcu7t5v0.spice
Is this supposed to 180MCU_STDCELLS for portability?

The approach of distributing currents to each unit cell is not commonly used. It is more compact to distribute the bias voltages (but this also needs good layout to avoid IR drop issues).

junbeomic replied:
what does "distribute the bias voltages" mean? I think that each current source(cascode current source at here) of unit cell is connected to current mirror from bandgap current source as a below schematic.
![image](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/346bc9b0-0146-46cd-ada2-7d28d932340a)

junbeomic replied:
and 180MCU_REF_7T5V0/spice/gf180mcu_fd_sc_mcu7t5v0.spice is used to use this one "${PDK_ROOT}/${PDK}/libs.ref/gf180mcu_fd_sc_mcu7t5v0"
is it better to change 180MCU_REF_7T5V0 with 180MCU_STDCELLS? 
in my xschemrc I set this value as 180MCU_REF_7T5V0 as below.
![image](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/031fdb9a-b7bb-41a8-a81f-37c8c14c3f1c)

bmurmann replied:
See e.g. here for a DAC schematic with bias voltage distribution: https://www.semanticscholar.org/paper/The-Current-Steering-DAC-[A-Circuit-for-All-Razavi/b0f046346e44480956fb5cddc0e8c1f183190a15/figure/1

bmurmann replied:
Perhaps others can comment on the path question (@d-m-bailey?). I am not sure what is the best way for portability.

bmurmann replied:
Unrelated -- can someone from team LTC2 respond to my question here?
https://open-source-silicon.slack.com/archives/C04MJUYP99V/p1698072782967519

We would like you to present an update next week.

junbeomic replied:
Thank you for suggesting bias voltage distribution.
and I will ask for comment about the path question.
and two weeks ago, I couldn't attend meet-up due to personal occasion so that I asked people about presenstation and they undestood Nov 8 is assigned.
Yes, each team(japan, chile, korea) will be okay that day. I understood that this schedule was fixed in meet-up two weeks ago :)

d-m-bailey replied:
`180MCU_REF_7T5V0` with `180MCU_STDCELLS` is most likely a matter of preference.
However, if you do use `180MCU_STDCELLS` it would be easy to make the change to a different track size.
Just change one line in the `xschemrc` file, as opposed to all references in all the schematics.

junbeomic replied:
Oh sorry, some people have not confirmed this schedule but we are checking this issues now.

junbeomic replied:
Everyone is OKAY that day :)

bmurmann replied:
> `180MCU_REF_7T5V0` with `180MCU_STDCELLS` is most likely a matter of preference. However, if you do use `180MCU_STDCELLS` it would be easy to make the change to a different track size. Just change one line in the `xschemrc` file, as opposed to all references in all the schematics.

Thanks @d-m-bailey. I see the pros and cons of each approach. If we use 180MCU_STDCELLS, I think each user needs to share a portable version of their xschemrc, so that it is well documented what library they are assuming. It sems to me that we should agree to some standards as a community, since effective sharing and reviewing is one of the main ideas of open source...

junbeomic replied:
@bmurmann prof.borris murmann I have a question about a bias voltage for current source. Could you give me an advice? it will be really helpful for me.

![2](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/8d024eef-cfd0-45ed-aeb9-c1136bf2d042)
I learned that a bias voltage from voltage divider is vulnerable to VDD fluctuation and PTV of current source. so people use a current mirror for better biasing. 
![image](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/6a84db76-352c-44fd-ad7e-e3c50f086b0f)
and I read some papers covering current steering DAC and most of them use bais schem as below.

![image](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/3a3a2fdb-189c-4d2f-a34b-884bca09c900)
(A 10-b, 500-MSample/s CMOS DAC in 0.6 mm Chi-Hung Lin and Klaas Bult)
Can I just use voltage divider for biasing? Thank you :)

bmurmann replied:
You should definitely not use a resistive voltage divider. My suggestion is to share the voltages VBP and VBC between multiple cells (instead of generating these voltages separately in each cell, which is somewhat impractical). Note that the paper you are citing does the same thing.

junbeomic replied:
Ah! I missed that point. Thank you very much

bmurmann reviewed (commented):
I can now easily open the cells, thank you. I have left some specific comments for the schematics that I had time to look at.

junbeomic commented:
I updated analog cells and added standard cells(latch) to complete LSB part schematics. but I have a worry about implementation of latch
In gf180mcu standard cell, there is no dual latch(having two complementary output(Q and negQ)). so I just adhered a output net with input(gate of transistors) of output inverter. but I am not sure it will be safe.
and to use this dual latch, users should modify gf180mcu_fd_sc_mcu7t5v0 file.
![image](https://github.com/sscs-ose/DC23-LTC2/assets/49300325/585219e8-3cdb-4286-8677-394e99275083)

d-m-bailey commented:
@aillish301 rather than modifying the pdk, I suggest putting this new cell in another location that people can access. Also remember to provide all the views (lef, timing, gds, etc.).

Also, changing `net5` to `QN` may be more intuitive.

msaligane reviewed (commented):
I don't think you need to push these files with your design. You can maybe add a readme on how you setup your environment so that we do not bloat the repo with extra files.

bmurmann reviewed (changes requested):
The diff pair devices in the DAC current unit cells are all the same size, but they should be scaled according to the unit currents. I.e., the 8x cell should use 8x larger diff pairs than the 1x cell. Generally, these transistors look too small to me. What is the Vgs-Vt of the diff pair transistors when steered left/right?

There is a symbol for dlxbp_1 but no schematic. Is this a cell from the GF library? Why is one of the outputs called net5?

Commit:
revised schematics, added layout of lsb matrix

Commit:
fixed file path
